src/hotspot/cpu/x86/x86.ad

```diff
@@ -10136,3 +10142,31 @@ instruct castVVLeg(legVec dst)
   ins_cost(0);
   ins_pipe(empty);
 %}
+
+instruct FloatClassCheck_reg_reg_vfpclass(rRegI dst, regF src, kReg ktmp, rFlagsReg cr)
+%{
+  predicate(VM_Version::supports_avx512dq());
```

Comment:
Predicate is no longer needed you have done the check in match_rule_supported routine.

Reply:
That's true, the check is already done in the match rule. Will update the code.